app/main.py

```diff
@@ -1 +1,35 @@
-# write your code here
+class Animal:
+
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}, " \
+               f"Health: {self.health}, " \
+               f"Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(target: Herbivore) -> None:
+        if not isinstance(target, Carnivore) and not target.hidden:
+            if target.health > 50:
```

Comment:
I think it would be better to simply subtract 50 from "target.health" attribute, then check: if this attribute is less than or equal to zero and if so, delete it. Then your method will have only two conditions (it is not necessary to use nested conditions) so the code will run faster and look cleaner.